v-next/hardhat-node-test-runner/src/protocol.ts

```diff
@@ -0,0 +1,148 @@
+import type { HardhatRuntimeEnvironment } from "hardhat/types/hre";
+
+import fs from "node:fs";
+import net from "node:net";
+import * as os from "node:os";
+
+import { assertHardhatInvariant } from "@nomicfoundation/hardhat-errors";
+
+function getSocketPath(): string {
+  if (os.platform() === "win32") {
+    return "\\\\.\\pipe\\hardhat_node_test";
+  } else {
+    return "/tmp/hardhat_node_test.sock";
```

Comment:
Same about the random. We may want to place this under the cache, or use mktmp from the ustils.

Reply:
Should I consider switching to tcp socket instead of file ? I don't think there are any downsides besides choosing an available port. Looks like using file sockets have some overhead